main.py

```
@@ -3,113 +3,165 @@
-        return []
+# -------------------- FETCH --------------------
+def get_github_repos(year, page):
+    """Fetch CVE repos from GitHub with retry."""
+    for attempt in range(RETRY_ATTEMPTS):
+        try:
+            resp = requests.get(GITHUB_API_URL.format(year, page), timeout=15)
+            resp.raise_for_status()
+            return resp.json().get("items", [])
+        except requests.RequestException as e:
+            logging.warning(f"GitHub attempt {attempt+1} failed: {e}")
+            time.sleep(random.randint(3, 8))
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)